### PR TITLE
FIX: Add href to post-date link element

### PR DESCRIPTION
**Pull request by nbianca**
The href was removed in commit 08a1f41582534c2793ca21d67df34ef6821b40d0,
but it was useful to quick copy the URL to the post.

### CODE REVIEW

**Review by pmusaraj — APPROVED** [review]

**Comment by pmusaraj** [comment]
Note that test failures seem relevant.

**Comment by ZogStriP** [comment]
@nbianca I think it's worth adding a test to ensure we document this feature.

File: app/assets/javascripts/discourse/app/widgets/hooks.js [hunk 1]
```diff
@@ -65,6 +65,21 @@ export const WidgetMouseOverHook = buildHook(MOUSE_OVER_ATTRIBUTE_NAME);
 export const WidgetMouseOutHook = buildHook(MOUSE_OUT_ATTRIBUTE_NAME);
 export const WidgetTouchEndHook = buildHook(TOUCH_END_ATTRIBUTE_NAME);
 
+export class WidgetClickElementHook extends WidgetBaseHook {
```
Comment (pmusaraj) [inline app/assets/javascripts/discourse/app/widgets/hooks.js:68]: @nbianca this is a bit more complicated than the initial commit. Can you explain why we need this extra complexity? It doesn't seem necessary.

**Comment by pmusaraj** [comment]
@nbianca the updated version of this PR is considerably more complicated, are you sure we need this extra complexity?

**Comment by nbianca** [comment]
I am afraid so, I could not find a solution with what we have right now.

The problem is that when I defined `click` then it did not work as expected because the event handler is registered on the document and then that handler passes it down to the element that registered it. This does not work with `a` elements because the method is not even called. 

The problem with `mouseup` is that there are many edge cases that the browser simply handles with `click` events. The extra code is to make sure that the event handler is registered on the right element.

**Comment by nbianca** [comment]
Worked a different solution in #16469.

**Pull request closed without merging**